From 30da35bbf05112c65a1a72f6c4bfa33614f44c40 Mon Sep 17 00:00:00 2001
From: brunduk <nikolai.viktorovich@gmail.com>
Date: Thu, 17 Sep 2026 23:53:23 -0300
Subject: [PATCH 2/2] d/wmr: Drop the extra wait at the end of the camera USB
 thread

The loop only ends once wmr_camera_free() has set usb_complete or
os_thread_helper_stop_and_wait() has cleared running, and the latter
signals once, before joining. If that signal lands while the thread is
still returning from libusb_handle_events_completed(), nothing wakes it
from this trailing wait and the join never returns. It guards no state.

Signed-off-by: brunduk <nikolai.viktorovich@gmail.com>
---
 src/xrt/drivers/wmr/wmr_camera.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/xrt/drivers/wmr/wmr_camera.c b/src/xrt/drivers/wmr/wmr_camera.c
index a798b3e53..e38ade2d0 100644
--- a/src/xrt/drivers/wmr/wmr_camera.c
+++ b/src/xrt/drivers/wmr/wmr_camera.c
@@ -258,9 +258,6 @@ wmr_cam_usb_thread(void *ptr)
 
 		os_thread_helper_lock(&cam->usb_thread);
 	}
-
-	//! @todo Think this is not needed? what condition are we waiting for?
-	os_thread_helper_wait_locked(&cam->usb_thread);
 	os_thread_helper_unlock(&cam->usb_thread);
 
 	return NULL;
-- 
2.47.3

